lib/active_admin/csv_builder.rb

```diff
@@ -105,7 +105,7 @@ def initialize(name, resource = nil, options = {}, block = nil)
 
       def humanize_name(name, resource, humanize_name_option)
         if humanize_name_option
-          name.is_a?(Symbol) && resource.present? ? resource.human_attribute_name(name) : name.to_s.humanize
+          name.is_a?(Symbol) && resource.respond_to?(:resource_class) ? resource.resource_class.human_attribute_name(name) : name.to_s.humanize
```

Comment:
This is the first thing that surprised me, I would expect that we shouldn't need the `respond_to?` check because our resources always respond to `resource_class`. It turns out that the `ResourceDSL` class is initialized with an AA resource object _and_ the underlying object (raw `Post`, for example), and this second object is what's used by the `CSVBuilder`.

Since we are refactoring, I think this dependency in the `ResourceDSL` class is not necessary at all (if it needs to access the underlying resource, it can always do `config.resource_class`). So I would kill the overriden initializer in https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/resource_dsl.rb#L4-L8 and use `config` instead of `@resource` here: https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/resource_dsl.rb#L110-L112.

That way we could keep the old `present?` check and just change `resource.human_attribute_name` with `resource.resource_class.human_attribute_name` since this class now gets the actual resource and not its class.

Makes sense?